README.md

```diff
@@ -202,44 +234,96 @@ the `lookup` result will sometimes provide a new array, and sometimes not.
   supported.
 * **label** [String][]: Optional label in which to store the result.
 
-## `execute()`
+**Returns:** `this<SetupChain>`
 
-Executes each task added sequentially and collects the results into a single object.
-The results for `execute` are returned. The internal state will be Persisted.
+```javascript
+const SetupChain = require('@logdna/setup-chain')
 
-**returns**: [Object][]
+class MyChain extends SetupChain {
+  constructor(state) {
+    super(state, {
+      hello: async () => {
+        return 'hi'
+      }
+    })
+  }
+}
 
-```javascript
-await new SetupChain()
-  .foo()
-  .bar()
-  .insertDBRecord({}, 'record')
+const chain = new MyChain()
+chain
+  .repeat(5, 'hello', {}, 'result')
   .execute()
+  .then((result) => {
+    console.log(result)
+  })
 
-{
-  foo: 'bar'
-, bar: 'foo'
-, record: { id: 1 }
-}
+> {result: ['hi', 'hi', 'hi', 'hi', 'hi']}
 ```
 
-## `sleep(<Object>)`
+## `sleep(opts)`
 
-### Parameters
+When used in a chain, it will wait for a specified number of milliseconds before returning. Since this is a chain action, the sleep will appear to happen sequentially
+within the action chain.
+
+**Parameters**
 
 * **opts** [Object][]: configuration options
   * **ms** [Number][]: The number of milliseconds to wait before continuing
 
-This is a builder function for the `sleep` action.
-It will Waits for a specified number of milliseconds before returning.
-
-**returns**: SetupChain
+**Returns**: `this<SetupChain>`
 
 ## `reset()`
 
 Manually clears internal state and any pending tasks
 
-**returns**: SetupChain
+**Returns**: `this<SetupChain>`
+
+## `execute()`
+
+Resolves the promises returned by chain actions, and returns the result
+of each action in a single object. This should be the final
+method in any chain, otherwise the actions will not be resolved into values.
+After `execute` is called, the return value is also stored in `chain.state` for
+access later, even if the return value is not immediately used.
+
+**Returns**: [Object][]
+
+## `lookup(input)`
+
+Resolves lookup value. This function is generally used by actions to resolve values
+from the chain as it is executes, thus this function **is not chainable**. To use
+this function within a chain, see examples in the [set action](#setstringobject-any).
```

Comment:
"To use this function within a chain..."

I don't think it's clear _why_ you would want to do this.

maybe:

> For a chainable action that persists a value to the chain state, use `set()` instead.

and then use one of the `set()` calls in the example below to demonstrate?

```js
.set('five', '#four.2')
```

Reply:
Yep, that one was messy.  Thanks!